src/urllib3/util/url.py

```diff
@@ -252,7 +251,7 @@ def idna_encode(name):
             *validator.COMPONENT_NAMES
         ).validate(uri_ref)
     except ValidationError:
-        six.raise_from(LocationParseError(url), None)
+        return six.raise_from(LocationParseError(url), None)
```

Comment:
Is the `return` here actually necessary?

Reply:
Not at all, however IDEs don't recognize that `six.raise_from()` won't allow execution to propagate. Should I remove it? Don't mind either way.